language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/common/utils/CommonUtil.java

```diff
@@ -1631,19 +1631,13 @@ public static boolean isSelfClassSymbol(Symbol symbol, PositionedOperationContex
      * @param nodeAtCursor         Node
      * @return {@link Boolean} whether the symbol is a self object symbol.
      */
-
     public static boolean isSelfObjectSymbol(Symbol symbol, Node nodeAtCursor) {
         Node currentNode = nodeAtCursor;
-        while (currentNode != null) {
-            if (currentNode.kind() == SyntaxKind.OBJECT_CONSTRUCTOR) {
-                break;
-            }
+        while (currentNode != null && currentNode.kind() != SyntaxKind.OBJECT_CONSTRUCTOR) {
             currentNode = currentNode.parent();
         }
-        if (currentNode == null || currentNode.kind() != SyntaxKind.OBJECT_CONSTRUCTOR) {
-            return false;
-        }
-        return symbol.getName().isPresent() && symbol.getName().get().equals(SELF_KW);
+        return currentNode != null && currentNode.kind() == SyntaxKind.OBJECT_CONSTRUCTOR
+                && symbol.getName().isPresent() && symbol.getName().get().equals(SELF_KW);
```

Comment:
you can use `symbol.getName().orElse("").equals(SELF_KW)` instead of `symbol.getName().isPresent() && symbol.getName().get().equals(SELF_KW)`